valve-monitor/web-app/src/components/elements/AlarmThresholdCard.tsx

```diff
@@ -0,0 +1,101 @@
+import { InfoCircleOutlined } from "@ant-design/icons";
+import { Card, Form, InputNumber, Tooltip } from "antd";
+import { updateAlarmThreshold } from "../../api-client/fleetVariables";
+import { ERROR_MESSAGE } from "../../constants/ui";
+import cardStyles from "../../styles/Card.module.scss";
+
+export interface AlarmThresholdProps {
+  currentMinFlowThreshold: number | undefined;
+  currentMaxFlowThreshold: number | undefined;
+  setCurrentMinFlowThreshold: (currentMinFlow: number | undefined) => void;
+  setCurrentMaxFlowThreshold: (currentMaxFlow: number | undefined) => void;
+  setIsErrored: (isErrored: boolean) => void;
+  setIsLoading: (isLoading: boolean) => void;
+  setErrorMessage: (errorMessage: string) => void;
+}
+
+const AlarmThresholdCard = (props: AlarmThresholdProps) => {
+  const {
+    currentMinFlowThreshold,
+    currentMaxFlowThreshold,
+    setCurrentMinFlowThreshold,
+    setCurrentMaxFlowThreshold,
+    setIsErrored,
+    setIsLoading,
+    setErrorMessage,
+  } = props;
+
+  const [form] = Form.useForm();
+
+  const update = async () => {
+    const min = form.getFieldValue("min") || undefined;
+    const max = form.getFieldValue("max") || undefined;
+
+    if (min === currentMinFlowThreshold && max === currentMaxFlowThreshold) {
+      return;
+    }
+
+    setIsErrored(false);
+    setErrorMessage("");
+    setIsLoading(true);
+
+    try {
+      await updateAlarmThreshold(min, max);
+      setCurrentMinFlowThreshold(min);
+      setCurrentMaxFlowThreshold(max);
+    } catch (e) {
+      setIsErrored(true);
+      setErrorMessage(ERROR_MESSAGE.UPDATE_FLEET_MONITOR_FREQUENCY_FAILED);
+    }
+    setIsLoading(false);
+  };
+
+  return (
+    <Card
+      className={cardStyles.cardContainer}
+      title={
+        <div className={cardStyles.cardTitle}>
+          <div>Alarm Threshold</div>
+        </div>
+      }
+    >
+      <div>
+        <p>
+          Configure default flow rate range
+          <Tooltip
+            color="#416681"
```

Comment:
I’m not proud of this, but setting this color in CSS in Ant ended up being a mess, and there are lots of confusing Stack Overflow posts about it. This just worked so I called it good enough for now.

Reply:
Fine by me. We had to do something similar with chart colors when using Chart.js.